src/command/subgraph/publish.rs

```diff
@@ -74,4 +126,84 @@ impl Publish {
             publish_response,
         })
     }
+
+    // output stream is optionally injected (defaulting to stdout) so we can test this function
+    pub fn handle_invalid_routing_url(
+        maybe_prompt_response: Option<bool>,
+        maybe_output: Option<&mut dyn std::io::Write>,
+    ) -> Option<RoverOutput> {
+        let default_output = &mut std::io::stdout();
```

Comment:
rover prints to `stderr` for warnings and progress logs, and `stdout` is only ever written to from `RoverOutput` - this is so machine readable output (like schemas) can be piped out to files or into other programs (`rover graph introspect <ENDPOINT> | rover graph publish <GRAPH_REF> --schema -`)

these warnings should be printed out to `stderr`

Reply:
i don't think this code actually needs to deal with `RoverOutput` at all since it won't be modifying that struct or needing to return it early, we're just printing some warnings and in the future erroring out